packages/go/stbernard/workspace/redoc/redoc.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package redoc
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/specterops/bloodhound/packages/go/stbernard/cmdrunner"
+	"github.com/specterops/bloodhound/packages/go/stbernard/environment"
+)
+
+func GenerateOpenAPIDoc(projectPath string, submodules []string, env environment.Environment) error {
+
+	// Either we are in the `bhce` submodule or we must find it
+	var basePath = projectPath
+	for _, submodule := range submodules {
+		if filepath.Base(submodule) == "bhce" {
+			basePath = submodule
+			break
+		}
+	}
+
+	var (
+		srcPath    = filepath.Join(basePath, "packages", "go", "openapi")
+		inputPath  = filepath.Join(srcPath, "src/openapi.yaml")
+		outputPath = filepath.Join(srcPath, "doc/openapi.json")
+		command    = "redocly"
```

Comment:
We should run the same npx command (with version constraint) here as we do for `gen-spec`. Otherwise, this will choke if you don't have redocly installed

Reply:
ah good catch, I forgot to make this update once i solved the npx issue i was having!